src/omv/ports/mimxrt/omv_gpu.c

```diff
@@ -0,0 +1,247 @@
+        if (src_img->w == src_rect->w) {
+            SCB_CleanDCache_by_Addr(src16, src_rect->w * src_rect->h * sizeof(uint16_t));
+        } else {
+            for (int i = 0; i < src_rect->h; i++) {
+                SCB_CleanDCache_by_Addr(src16, src_rect->w * sizeof(uint16_t));
+                src16 += src_img->w;
+            }
+        }
+        #endif
+    }
+
+    PXP_SetProcessSurfaceBufferConfig(PXP, &input_buffer_config);
+    PXP_SetProcessSurfaceScaler(PXP, src_rect->w, src_rect->h, dst_rect->w, dst_rect->h);
+    PXP_SetProcessSurfacePosition(PXP, 0, 0, dst_rect->w - 1, dst_rect->h - 1);
+
+    pxp_output_buffer_config_t output_buffer_config = {};
+
+    if (dst_img->pixfmt == PIXFORMAT_GRAYSCALE) {
+        uint8_t *dst8 = IMAGE_COMPUTE_GRAYSCALE_PIXEL_ROW_PTR(dst_img, dst_rect->y) + dst_rect->x;
+
+        output_buffer_config.pixelFormat = kPXP_OutputPixelFormatY8;
+        output_buffer_config.buffer0Addr = (uint32_t) dst8;
+        output_buffer_config.pitchBytes = dst_img->w;
+
+        #if __DCACHE_PRESENT
+        // Ensures any cached writes to dst8 are flushed.
+        if (dst_img->w == dst_rect->w) {
+            SCB_CleanInvalidateDCache_by_Addr(dst8, dst_rect->w * dst_rect->h);
+        } else {
+            for (int i = 0; i < dst_rect->h; i++) {
+                SCB_CleanInvalidateDCache_by_Addr(dst8, dst_rect->w);
+                dst8 += dst_img->w;
+            }
+        }
+        #endif
+    } else {
+        uint16_t *dst16 = IMAGE_COMPUTE_RGB565_PIXEL_ROW_PTR(dst_img, dst_rect->y) + dst_rect->x;
+
+        output_buffer_config.pixelFormat = kPXP_OutputPixelFormatRGB565;
+        output_buffer_config.buffer0Addr = (uint32_t) dst16;
+        output_buffer_config.pitchBytes = dst_img->w * sizeof(uint16_t);
+
+        PXP_EnableCsc1(PXP, src_img->pixfmt != PIXFORMAT_RGB565);
+
+        #if __DCACHE_PRESENT
+        // Ensures any cached writes to dst16 are flushed.
+        if (dst_img->w == dst_rect->w) {
+            SCB_CleanInvalidateDCache_by_Addr(dst16, dst_rect->w * dst_rect->h * sizeof(uint16_t));
+        } else {
+            for (int i = 0; i < dst_rect->h; i++) {
+                SCB_CleanInvalidateDCache_by_Addr(dst16, dst_rect->w * sizeof(uint16_t));
+                dst16 += dst_img->w;
+            }
+        }
+        #endif
+    }
+
+    output_buffer_config.width = dst_rect->w;
+    output_buffer_config.height = dst_rect->h;
+
+    PXP_SetOutputBufferConfig(PXP, &output_buffer_config);
+
+    pxp_flip_mode_t flip_mode = kPXP_FlipDisable;
+    flip_mode |= (hint & IMAGE_HINT_HMIRROR) ? kPXP_FlipHorizontal : 0;
+    flip_mode |= (hint & IMAGE_HINT_VFLIP) ? kPXP_FlipVertical : 0;
+    PXP_SetRotateConfig(PXP, kPXP_RotateProcessSurface, kPXP_Rotate0, flip_mode);
+
+    PXP_ClearStatusFlags(PXP, kPXP_CompleteFlag);
+    PXP_Start(PXP);
+
+    while (!(kPXP_CompleteFlag & PXP_GetStatusFlags(PXP))) {
+        MICROPY_EVENT_POLL_HOOK
+    }
+
+    PXP_Reset(PXP);
+    PXP_Deinit(PXP);
```

Comment:
Maybe initialize once and just leave it initialized ?

Reply:
It crashes and locks up the CPU if you do.